cli/arguments/reference.go

```diff
@@ -69,20 +73,39 @@ func ParseReference(arg string, parseArch bool) (*Reference, error) {
 		ret.Version = toks[1]
 	}
 
-	if parseArch {
-		toks = strings.Split(ret.PackageName, ":")
-		if len(toks) != 2 {
-			return nil, fmt.Errorf(tr("invalid item %s"), arg)
-		}
-		if toks[0] == "" {
-			return nil, fmt.Errorf(tr("invalid empty core name '%s'"), arg)
+	toks = strings.Split(ret.PackageName, ":")
+	if len(toks) != 2 {
+		return nil, fmt.Errorf(tr("invalid item %s"), arg)
+	}
+	if toks[0] == "" {
+		return nil, fmt.Errorf(tr("invalid empty core name '%s'"), arg)
+	}
+	ret.PackageName = toks[0]
+
+	// try to search in the packages, this way ParseReference can be case insensitive
+	// note that if the search fail for some reason the ret.Packagename does not get overwritten
+	var targetPackage *cores.Package
+	if inst != nil { // do not perform the search if the instance is not defined
+		pm := commands.GetPackageManager(inst.Id)
+		for _, targetPackage = range pm.Packages {
+			if targetPackage.Name == strings.ToLower(ret.PackageName) {
+				ret.PackageName = targetPackage.Name
+				break
+			}
 		}
-		ret.PackageName = toks[0]
-		if toks[1] == "" {
-			return nil, fmt.Errorf(tr("invalid empty core architecture '%s'"), arg)
+	}
+	if toks[1] == "" {
+		return nil, fmt.Errorf(tr("invalid empty core architecture '%s'"), arg)
+	}
+	ret.Architecture = toks[1]
+	// we do the same for architecture
+	if inst != nil { // do not perform the search if the instance is not defined
+		for _, platform := range targetPackage.Platforms {
+			if platform.Architecture == strings.ToLower(ret.Architecture) {
```

Comment:
```suggestion
			if strings.ToLower(platform.Architecture) == strings.ToLower(ret.Architecture) {
```
Architecture names are not always lower case.

For example:
https://github.com/redbear/redbear.github.io/blob/master/arduino/package_redbear_nRF5x_index.json#L14
```
                    "architecture": "nRF51822",
```

Reply:
using [`strings.EqualFold`](https://pkg.go.dev/strings#EqualFold) should be sufficient. Thanks Per